--- configure.ac
+++ configure.ac
@@ -247,7 +247,7 @@ AC_CACHE_CHECK([for libevent directory], ac_cv_libevent_dir, [
          AS_IF(test "$SUNCC" = "yes" -o "x$_myos" = "xsolaris2",
                [saved_LDFLAGS="$saved_LDFLAGS -Wl,-R$ledir/lib"],
                [AS_IF(test "$GCC" = "yes",
-                     [saved_LDFLAGS="$saved_LDFLAGS -Wl,-rpath=$ledir/lib"])])
+                     [saved_LDFLAGS="$saved_LDFLAGS -Wl,-rpath,$ledir/lib"])])
        else
          ac_cv_libevent_dir="(system)"
        fi

